### PR TITLE
Remote groups via HTTP Headers

### LINKED ISSUE #5775: Add the ability to Sync netbox user Groups and special Privileges with an Remote (HTTP) Auth Backend
### Environment
* Python version:  3.8.x
* NetBox version:  2.10.4
altough both n/a

### Proposed Functionality
Adding Group sync functionality to the Remote (HTTP) Auth Backend 
I have a working implementation under https://github.com/MaxRink/netbox/tree/remote_groups

### Use Case
Unlike the LDAP Backend, the RemoteUserBackend wasnt able to Sync Group memberships form the auth Provider.
This brings it up to parity, as the new Settings give you the Option to setup an automatic Group Sync (and revocation) on each User login.
This has been tested with oauth2-proxy
The old behaviopur of just Syncing Users and then manually managing Groups should be unaffected.

### Database Changes
No

### External Dependencies
No, all Django-internal

### PR DESCRIPTION
**Pull request by MaxRink**
### Fixes: #5775

Unlike the LDAP Backend, the RemoteUserBackend wasnt able to Sync Group memberships form the auth Provider.
This brings it up to parity, as the new Settings give you the Option to setup an automatic Group Sync (and revocation) on each User login.
This has been tested with [oauth2-proxy](https://github.com/oauth2-proxy/oauth2-proxy)
The old behaviopur of just Syncing Users and then manually managing Groups should be unaffected.
ToDo: 
 - [ ] Tests for new Settings

### CODE REVIEW

**Comment by jeremystretch** [comment]
Thank you for your interest in contributing to NetBox, however it appears there is no accepted issue that correlates to this pull request.  Before pull requests are opened, we require an accepted issue as per our [contributing guide](https://github.com/digitalocean/netbox/blob/master/CONTRIBUTING.md).  Please first open an issue and wait for it to be accepted before further work is done on this pull request.

**Comment by MaxRink** [comment]
@jeremystretch there is :D 
https://github.com/netbox-community/netbox/issues/5775#issuecomment-874792743

**Comment by jeremystretch** [comment]
Ok, I see that. For future, please be sure to retain the `### Fixes: #ISSUENUMBER` header from the PR template. This ensures that issues are correctly correlated with PRs and automatically closed when a PR is merged. It makes my job a lot easier. :slightly_smiling_face:

File: docs/configuration/optional-settings.md [hunk 1]
```diff
@@ -472,6 +480,54 @@ When remote user authentication is in use, this is the name of the HTTP header w
...
+
+## REMOTE_AUTH_STAFF_USERS
+
+Default: `[]` (Empty list)
+
+The list of users that get promoted to Staff on Login. If user isn't present in list on next Login, the Role gets revoked. (Requires `REMOTE_AUTH_ENABLED` and `REMOTE_AUTH_GROUP_SYNC_ENABLED` )
+
+---
+
+## REMOTE_AUTH_GROUP_SEPERATOR
```
Comment (nolta) [inline docs/configuration/optional-settings.md:523]: Minor typo: should be `REMOTE_AUTH_GROUP_SEPARATOR`

Comment (MaxRink) [reply docs/configuration/optional-settings.md]: Good catch, thx

**Comment by jeremystretch** [comment]
Can we get some remote authentication users to test this and weigh in here?

**Comment by mackaybe** [comment]
Successfully tested with oauth2-proxy and keycloak

**Comment by davama** [comment]
Just noticed this

will test with http header auth
Will update tomorrow 👍👍
Thank you!

**Comment by sol1-matt** [comment]
I've patched the 3.0.2 release tag with this diff on our dev box and SSO login continues to work.
No problems found in the upgrade.

I say continues as we patched the previous version (2.11) as well, not sure if that previous patch was the same as this one or different as the person who did it said it was 'based on it'.

**Comment by jeremystretch** [comment]
@davama were you able to test this? Can you share your findings?

**Comment by davama** [comment]
@jeremystretch @mackaybe 
applied this patch to my v3.0.0 instance ... i would assume it would also work on v3.0.2 as @sol1-matt mentioned

works good :+1: 
using these settings.
REMOTE_HTTP_GROUPS syncs with netbox.

```
REMOTE_AUTH_ENABLED = True
REMOTE_AUTH_BACKEND = 'netbox.authentication.RemoteUserBackend'
REMOTE_AUTH_HEADER = 'HTTP_REMOTE_USER'
REMOTE_AUTH_AUTO_CREATE_USER = True
REMOTE_AUTH_DEFAULT_GROUPS = ['netbox']
REMOTE_AUTH_DEFAULT_PERMISSIONS =  { "is_superuser" }
REMOTE_AUTH_GROUP_SYNC_ENABLED = True
REMOTE_AUTH_GROUP_HEADER = 'HTTP_REMOTE_GROUPS'
REMOTE_AUTH_SUPERUSER_GROUPS = ['netbox'] 
REMOTE_AUTH_STAFF_GROUPS = ['netbox']
REMOTE_AUTH_GROUP_SEPARATOR = ','
```

Would be awesome if `email` , `first_name` , `last_name` would sync too... but outside this PR

**Pull request merged**